File: sales-by-match/sales-by-match.js
// https://www.hackerrank.com/challenges/sock-merchant/problem

const ar = [10, 20, 20, 10, 10, 30, 50, 10, 20]

function count(arr, item) {
    let count = {}
    for (let i of arr) {
        count[i] = (count[i] || 0) + 1
    }
    return count[item]
}

let res = 0
for (let item of [... new Set(ar)]) {
    res += Math.floor(count(ar,item) / 2)
}

console.log(res);

File: sales-by-match/sales-by-match.py
# https://www.hackerrank.com/challenges/sock-merchant/problem

ar = [10, 20, 20, 10, 10, 30, 50, 10, 20]

res = 0
for i in set(ar):
    res += ar.count(i) / 2
    
print(res)